src/ripple/app/misc/Validations.cpp

```diff
@@ -387,6 +387,39 @@ class ValidationsImp : public Validations
         return ret;
     }
 
+    std::uint32_t getValidationTime (uint256 const& hash, int minValidations)
```

Comment:
Completely missing the point with respect to the return type. A zero as a sentinel is weak because you can't tell by looking at the signature that there's a magic value representing failure:
```
std::uint32_t getValidationTime (uint256 const& hash, int minValidations)
```
versus
```
boost::optional<std::uint32> getValidationTime (uint256 const& hash, int minValidations)
```

We added the function `getValidationTime`, but we forgot to document that it returns 0 to indicate that there is no validation time:
https://github.com/ripple/rippled/commit/2fee2d91296c18312346fd9ac93aade0d7c92248#diff-dac56fbd1f44ebab8faacd41bd6d4687R67

We could have written this instead:
```
/** Returns the time a ledger was validated.
    @param ledger Hash of the ledger to look up
    @param minValidations The required number of validations
    @return The Ripple time at least `minValidations` validations for `ledger` was
            received, or boost::none if there were not enough validations.
*/
virtual
boost::optional<std::uint32_t>
getValidationTime (uint256 const& ledger, int minValidations) = 0;
```